File: Replace_Error_Code_with_Exception.py
# After
def withdraw(self, amount):
    if amount > self.balance:
        return -1
    else:
        self.balance -= amount
    return 0

# Refactoring
def withdraw(self, amount):
    if amount > self.balance:
        raize BalanceException()
    self.balance -= amount

File: Replace_Error_Code_with_Exception.js
// After
function withdraw(self, amount) {
    if (amount > self.balance) {
      return -1;
    } else {
      self.balance -= amount;
    }  
    return 0;
}    

// Refactoring
function withdraw(self, amount) {
    if (amount > self.balance) {
      throw BalanceException();
    }  
    self.balance -= amount;
}    
